### PR TITLE
[Feature] Do not allow 0 points for a quiz

### LINKED ISSUE #2852: 0 points for single quiz questions in exams should not be possible
### Is your feature request related to a problem?

As long as the overall quiz contains at least 1 point, it is possible to add (single) quiz questions with 0 points. 

### Describe the solution you'd like

As this is not needed for any type of exam, this option should be removed for exam exercises to prevent accidental fault during the exam creation and to not confuse students during the exam.
For consistency reasons, it would mostly make sense to just disable the "save" button when there are questions with 0 points.

### Describe alternatives you've considered

Disabling the save button in this case should do it.

### Additional context

image

### PR DESCRIPTION
**Pull request by derLalla**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

We don't want to have quizzes which allow 0 points for a quiz.
Fixes #2852 

### Description

We set the minimum points for each quiz to be 1 point instead of 0.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a quiz exercise and try to set the points of a quiz exercise to 0 or below
4. The quiz should only be possible to save if the points of all quizzes are above 0.

### Test Coverage

No changes.

### Screenshots

Quiz cannot be saved if points are below 1: 
image

### CODE REVIEW

**Review by airelawaleria — COMMENTED** [review]
image
image

I get this when saving a quiz with 1 exercise with 1 point and second exercise woth 0 points. But the save button in this case is still enabled and no errors are detected.

**Review by JohannesWeiss — COMMENTED** [review]
I observed the same behaviour as @airelawaleria. Additionally, only the first two questions get marked red with the error, but a third one (with 0 points) does not get that error. When using -1 instead of 0, the save button was disabled correctly.

**Comment by derLalla** [comment]
> image
> image
> 
> I get this when saving a quiz with 1 exercise with 1 point and second exercise woth 0 points. But the save button in this case is still enabled and no errors are detected.

oh thanks! I have fixed that in the latest commit.

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts [hunk 1]
```diff
@@ -666,10 +666,8 @@ export class QuizExerciseDetailComponent implements OnInit, OnChanges, Component
             this.quizExercise.quizQuestions != undefined &&
             !!this.quizExercise.quizQuestions.length;
         const areAllQuestionsValid = this.quizExercise.quizQuestions?.every(function (question) {
-            if (question.points == undefined) {
-                return false;
-            }
-            if (question.points && question.points < 0) {
+            console.log('question.points', question.points);
```
Comment (FuchsDominik) [inline src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts:669]: Was this for debug purposes?

**Review by FuchsDominik — CHANGES_REQUESTED** [review]
Code looks good. Found a console log which was probably forgotten

File: src/main/webapp/i18n/en/quizExercise.json [hunk 2]
```diff
@@ -161,7 +161,6 @@
                 "noQuestion": "Quiz: The quiz has no question.",
                 "invalidStartTime": "Quiz: The Start Time is not valid.",
                 "startTimeInPast": "Quiz: The Start Time lies in the past.",
-                "quizZeroPoints": "Quiz: The total number of points is 0.",
                 "questionTitle": "Question {{index}}: The title is missing.",
                 "questionScore": "Question {{index}}: The points are invalid.",
```
Comment (stefanwaldhauser) [inline src/main/webapp/i18n/en/quizExercise.json:165]:
```suggestion
                "questionScore": "Question {{index}}: At least one point per question required.",
```

File: src/main/webapp/i18n/de/quizExercise.json [hunk 3]
```diff
@@ -161,7 +161,6 @@
                 "noQuestion": "Quiz: Das Quiz hat keine Fragen.",
                 "invalidStartTime": "Quiz: Die Startzeit ist nicht gültig.",
                 "startTimeInPast": "Quiz: Die Startzeit liegt in der Vergangenheit.",
-                "quizZeroPoints": "Quiz: Die Gesamtpunktzahl ist 0.",
                 "questionTitle": "Frage {{index}}: Der Titel fehlt.",
                 "questionScore": "Frage {{index}}: Ungültige Punktzahl.",
```
Comment (stefanwaldhauser) [inline src/main/webapp/i18n/de/quizExercise.json:165]:
```suggestion
                "questionScore": "Frage {{index}}: Mindestens einen Punkt pro Aufgabe erforderlich.",
```

**Review by stefanwaldhauser — DISMISSED** [review]
Looks Good except for the above-mentioned console log statement

File: src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts (hunk 1 above)
Comment (derLalla) [reply src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts:669]: 😅 thanks!

**Review by FuchsDominik — APPROVED** [review]
Approve regarding the code after Log got removed

**Review by tiffca — APPROVED** [review]
Tested successfully, can't set points < 0 and can only save if all points sum up to > 0 

Am I supposed to receive an error if I try to save the following: 
1 quiz 
- 1 MC quiz question with 2 points 
- 1 SA quiz question with 0 points 
throws error on save -> POST https://artemistest.ase.in.tum.de/api/quiz-exercises 400

image

**Review by airelawaleria — APPROVED** [review]

**Review by marvinchimeka — APPROVED** [review]
works as expected

**Review by simonzilker — APPROVED** [review]

**Review by stefanwaldhauser — APPROVED** [review]
Reapprove

**Review by JohannesWeiss — APPROVED** [review]
Re-Approve as well.

**Review by krusche — APPROVED** [review]

**Pull request merged**